### PR TITLE
Per 208, 206, 216

### PR DESCRIPTION
**Pull request by evanhackett**
Lots of changes here. Sorry for not breaking this into multiple branches with multiple PRs.

I did break things into multiple commits though.

The main things are:

added ability to rename captions.
added text input to caption form, so now a text string can be entered and saved for a caption.
added delete buttons for caption
minor styling tweaks
minor refactor of the url/route structure (this was necessary to make the caption editor page work)

### CODE REVIEW

**Review by JosephJindrich — CHANGES_REQUESTED** [review]
The delete caption page is still bugged. After messing around with it, I was able to make captions completely stop appearing when I went to the url page. I would test the delete function a bit more. If I have time I will try and figure it out.

File: src/client/src/views/CaptionForm.js [hunk 1]
```diff
@@ -1,11 +1,43 @@
...
 module.exports = {
+  oninit: function(vnode) {
+    // first set the current Scene, then set the current caption.
+    Scene.setCurrentScene(vnode.attrs.sceneId - 1)
+    Scene.setCurrentCaption(vnode.attrs.captionId - 1)
+  },
   view: function(vnode) {
     return m('', [
-      m('h1', `Caption ${vnode.attrs.id}`),
+      m('h1', Scene.currentCaption.name ? Scene.currentCaption.name : `Caption ${vnode.attrs.captionId}`),
```
Comment (JosephJindrich) [inline src/client/src/views/CaptionForm.js]: Change this to `m('h1', Scene.currentCaption.name ? Scene.currentCaption.name : 'Caption String Name'),` please

**Comment by JosephJindrich** [comment]
@evanhackett If you could, I'd like you to also change caption to caption string. That was the name we decided to give what we've been labeling captions i the code. I'd like to keep the same naming conventions through the project.

File: src/client/src/index.js [hunk 2]
```diff
@@ -19,12 +19,12 @@ const CaptionForm = require('./views/CaptionForm')
 m.route(document.body, '/start', {
   '/start': Start,
   '/scenes': Scenes,
-  '/edit-scene/:id': {
+  '/scenes/scene-:id': {
```
Comment (JosephJindrich) [inline src/client/src/index.js]: Could you change :id to :sceneId like it is in the caption section

File: src/client/src/views/SceneForm.js [hunk 3]
```diff
@@ -9,13 +9,13 @@ module.exports = {
...
-    const captions = Scene.current.captions
-
+    const captions = Scene.currentScene.captions
+    
     return m('', [
       m(m.route.Link, {
         href: '/scenes',
       }, 'Back To Scene List'),
-      m('h1', Scene.current.name ? Scene.current.name : `Scene ${Scene.current.id}`),
+      m('h1', Scene.currentScene.name ? Scene.currentScene.name : `Scene ${Scene.currentScene.id}`),
```
Comment (JosephJindrich) [inline src/client/src/views/SceneForm.js:18]: Make this `m('h1', 'Scene Name'),` please

File: src/client/src/views/SceneForm.js [hunk 4]
```diff
@@ -25,34 +25,46 @@ module.exports = {
       }, [
         m("input.new-name-input[type=text]", {
           oninput: function (e) {
-            Scene.current.name = e.target.value
+            Scene.currentScene.name = e.target.value
           },
-          value: Scene.current.name ? Scene.current.name : `Scene ${Scene.current.id}`
+          value: Scene.currentScene.name ? Scene.currentScene.name : `Scene ${Scene.currentScene.id}`
```
Comment (JosephJindrich) [inline src/client/src/views/SceneForm.js:30]: Make this `value: Scene.currentScene.name ? Scene.currentScene.name : ''` please

File: src/client/src/views/CaptionForm.js [hunk 5]
```diff
@@ -1,11 +1,43 @@
...
+          e.preventDefault()
+          Scene.saveCaptionName()
+          Scene.saveCaptionText()
+        }
+      }, [
+        m("input.new-name-input[type=text]", {
+          oninput: function (e) {
+            Scene.currentCaption.name = e.target.value
+          },
+            value: Scene.currentCaption.name ? Scene.currentCaption.name : `Caption ${Scene.currentCaption.id}`
```
Comment (JosephJindrich) [inline src/client/src/views/CaptionForm.js:26]: Change this to `value: Scene.currentCaption.name ? Scene.currentCaption.name : ''` please

File: src/client/src/index.js (hunk 2 above)
Comment (evanhackett) [reply src/client/src/index.js]: Ok I fixed this.

**Review by JosephJindrich — APPROVED** [review]
Looks good

**Pull request merged**